File: src/float.erl
-module(float).
-export([encode_float64/1]).

% Encode a float as MessagePack float64 format (0xcb followed by 8 bytes in big-endian)
encode_float64(Value) ->
    % IEEE 754 double-precision binary representation
    Bin = <<Value:64/float>>,
    % Prepend the MessagePack format byte 0xcb for float64
    <<16#cb, Bin/binary>>.

File: src/glepack/float.mjs
export function encodeFloat64(value) {
    const buffer = new ArrayBuffer(9);
    const view = new DataView(buffer);
    
    // Set the format byte
    view.setUint8(0, 0xcb);
    
    // Set the float value
    view.setFloat64(1, value, false); // false = big-endian
    
    return new Uint8Array(buffer);
  }